File: Data Structures & Algorithms/binary-search/submission-1.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        if len(nums) == 0:
            return -1
        left = 0
        right = len(nums) - 1
        while left <= right:
            mid = (left + right) // 2
            if nums[mid] == target:
                return mid
            elif nums[mid] > target:
                right -= 1
            else:
                left += 1
        return -1
    

File: Data Structures & Algorithms/jump-game/submission-1.cs
public class Solution {
    public bool CanJump(int[] nums) {
        var n = nums.Length;
        var goal = nums[n-1];
        for (var i=n-1; i>=0; i--)
        {
            if (nums[i] + i >= goal)
            {
                goal = i;
            }
        }
        return goal == 0;
    }
}
